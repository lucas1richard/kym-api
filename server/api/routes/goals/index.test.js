const { expect, use } = require('chai');
const { jestSnapshotPlugin } = require('mocha-chai-jest-snapshot');

describe('goals routes', () => {
  use(jestSnapshotPlugin());
  const routePrefix = '/api/goals';

  describe(`GET ${routePrefix}/meals/v1`, () => {
    before(globals.seedTestData);
    after(globals.destroyAllHook);

    it('creates goals and responds with the new goals', async () => {
      const goal = { protein: 20, carbs: 30, fat: 10 };
      const res = await globals.agent.postWithToken(`${routePrefix}/meals/v1`)
        .send({
          TRAIN: [goal, goal, goal, goal, goal, goal],
          REST: [goal, goal, goal, goal, goal, goal],
        });
      expect(res.statusCode).equal(201);
      expect(res.body).toMatchSnapshot();
    });
    it('validates the req', async () => {
      const goal = { carbs: 30, fat: 10 };
      const res = await globals.agent.postWithToken(`${routePrefix}/meals/v1`)
        .send({
          TRAIN: [goal, goal, goal, goal, goal, goal],
          REST: [goal, goal, goal, goal, goal, goal],
        });
      expect(res.statusCode).equal(400);
      expect(res.body).toMatchSnapshot();
    });
  });
});
