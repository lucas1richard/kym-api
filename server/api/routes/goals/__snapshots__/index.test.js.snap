// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`goals routes GET /api/goals/meals/v1 gets meal goals 1`] = `
Array [
  Object {
    "goals": Object {
      "REST": Array [
        Object {
          "carbs": 29.3,
          "fat": 13.1,
          "protein": 33.3,
        },
        Object {
          "carbs": 29.3,
          "fat": 13.1,
          "protein": 33.3,
        },
        Object {
          "carbs": 77.8,
          "fat": 34.7,
          "protein": 88.2,
        },
        Object {
          "carbs": 0,
          "fat": 0,
          "protein": 0,
        },
        Object {
          "carbs": 64.6,
          "fat": 28.8,
          "protein": 73.2,
        },
        Object {
          "carbs": 0,
          "fat": 0,
          "protein": 0,
        },
      ],
      "TRAIN": Array [
        Object {
          "carbs": 79.5,
          "fat": 29.9,
          "protein": 63.3,
        },
        Object {
          "carbs": 0,
          "fat": 0,
          "protein": 0,
        },
        Object {
          "carbs": 119.2,
          "fat": 44.8,
          "protein": 95,
        },
        Object {
          "carbs": 0,
          "fat": 0,
          "protein": 0,
        },
        Object {
          "carbs": 87.4,
          "fat": 32.8,
          "protein": 69.7,
        },
        Object {
          "carbs": 0,
          "fat": 0,
          "protein": 0,
        },
      ],
    },
    "id": 1,
    "user_uuid": "60f24265-ef45-4891-b4ef-e8fad4435308",
  },
]
`;

exports[`goals routes POST /api/goals/meals/v1 creates goals and responds with the new goals 1`] = `
Object {
  "goals": Object {
    "REST": Array [
      Object {
        "carbs": 30,
        "fat": 10,
        "protein": 20,
      },
      Object {
        "carbs": 30,
        "fat": 10,
        "protein": 20,
      },
      Object {
        "carbs": 30,
        "fat": 10,
        "protein": 20,
      },
      Object {
        "carbs": 30,
        "fat": 10,
        "protein": 20,
      },
      Object {
        "carbs": 30,
        "fat": 10,
        "protein": 20,
      },
      Object {
        "carbs": 30,
        "fat": 10,
        "protein": 20,
      },
    ],
    "TRAIN": Array [
      Object {
        "carbs": 30,
        "fat": 10,
        "protein": 20,
      },
      Object {
        "carbs": 30,
        "fat": 10,
        "protein": 20,
      },
      Object {
        "carbs": 30,
        "fat": 10,
        "protein": 20,
      },
      Object {
        "carbs": 30,
        "fat": 10,
        "protein": 20,
      },
      Object {
        "carbs": 30,
        "fat": 10,
        "protein": 20,
      },
      Object {
        "carbs": 30,
        "fat": 10,
        "protein": 20,
      },
    ],
  },
  "id": 6037,
  "user_uuid": "60f24265-ef45-4891-b4ef-e8fad4435308",
}
`;

exports[`goals routes POST /api/goals/meals/v1 validates the req 1`] = `
Array [
  Object {
    "context": Object {
      "key": "protein",
      "label": "protein",
    },
    "message": "\\"protein\\" is required",
    "path": Array [
      "TRAIN",
      0,
      "protein",
    ],
    "type": "any.required",
  },
]
`;
